src/sage/rings/species.py

```diff
@@ -0,0 +1,433 @@
+    
+    def _element_constructor_(self, x):
+        r"""
+        Construct the `k`-variate molecular species with the given data.
+
+        INPUT:
+
+        - ``x`` - an element of ``self`` or a tuple ``(H, M)`` where `H` is
+        the permutation group representation for the species and `M` is a
+        ``dict`` mapping each element of the domain of `H` to integers in
+        `\{ 1 \ldots k \}`, representing the set to which the element belongs.
+        """
+        if parent(x) == self:
+            return x
+        H, M = x
+        domain_partition = H.disjoint_direct_product_decomposition()
+        term = self._indices.one()
+        for part in domain_partition:
+            term *= self._indices.gen(self._project(H, M, part))
+        return self._from_dict({term: 1})
+
+    def __getitem__(self, x):
+        r"""
+        Calls _element_constructor_ on x.
+
+        TESTS::
+
+            sage: P = PolynomialMolecularDecomposition(1)
+            sage: At1 = AtomicSpecies(1)
+            sage: At1((SymmetricGroup(1), {1: 1})).rename("X")
+            sage: X2 = (SymmetricGroup(2).young_subgroup([1, 1]), {1: 1, 2: 1})
+            sage: P[X2]
+            X^2
+            sage: P2 = PolynomialSpecies(2)
+            sage: At2 = AtomicSpecies(2)
+            sage: At2((SymmetricGroup(1), {1: 1})).rename("X")
+            sage: At2((SymmetricGroup(1), {1: 2})).rename("Y")
+            sage: XY = (SymmetricGroup(2).young_subgroup([1, 1]), {1: 1, 2: 2})
+            sage: P2[XY]
+            X*Y
+        """
+        return self._element_constructor_(x)
+
+    @cached_method
+    def one_basis(self):
+        r"""
+        Returns SymmetricGroup(0), which indexes the one of this algebra,
+        as per :meth:`AlgebrasWithBasis.ParentMethods.one_basis`.
+
+        EXAMPLES::
+
+            sage: P = PolynomialSpecies(1)
+            sage: P.one_basis()
+            1
+            sage: P2 = PolynomialSpecies(2)
+            sage: P2.one_basis()
+            1
+        """
+        return self._indices.one()
+
+    @cached_method
+    def an_element(self):
+        """
+        Return an element of ``self``.
+
+            sage: P = PolynomialSpecies(1)
+            sage: P.an_element()
+            1
+            sage: P2 = PolynomialSpecies(2)
+            sage: P2.an_element()
+            1
+        """
+        return self.one()
+
+    def product_on_basis(self, H, K):
+        r"""
+        Return the product of the basis elements indexed by `H` and `K`.
+
+        EXAMPLES::
+
+            sage: P=PolynomialMolecularDecomposition(1)
+            sage: d2 = {e: 1 for e in range(1, 3)}
+            sage: d3 = {e: 1 for e in range(1, 4)}
+            sage: L1 = [(H, d3) for H in SymmetricGroup(3).conjugacy_classes_subgroups()]
+            sage: L2 = [(H, d2) for H in SymmetricGroup(2).conjugacy_classes_subgroups()]
+            sage: matrix([[P(x) * P(y) for x in L1] for y in L2])
```

Comment:
```suggestion
            sage: A = AtomicSpecies(1)
            sage: A((SymmetricGroup(1), {1: 1})).rename("X")
            sage: [A((SymmetricGroup(n), {e: 1 for e in range(1, n+1)})).rename(f"E_{n}") for n in range(2, 5)]
            [None, None, None]
            sage: [A((CyclicPermutationGroup(n), {e: 1 for e in range(1, n+1)})).rename(f"C_{n}") for n in range(3, 5)]
            [None, None]

            sage: P = PolynomialSpecies(1)
            sage: d2 = {e: 1 for e in range(1, 3)}
            sage: d3 = {e: 1 for e in range(1, 4)}
            sage: L1 = [P((H, d3)) for H in SymmetricGroup(3).conjugacy_classes_subgroups()]
            sage: L2 = [P((H, d2)) for H in SymmetricGroup(2).conjugacy_classes_subgroups()]
            sage: matrix([[F * G for F in L1] for G in L2])
            [    X^5 X^3*E_2 C_3*X^2 E_3*X^2]
            [X^3*E_2 X*E_2^2 C_3*E_2 E_3*E_2]
```
It would also be good to have a shortcut for univariate species, such that passing the dictionary is not necessary in this case.

Reply:
Done.